Mathlib/MeasureTheory/Group/FundamentalDomain.lean

```diff
@@ -718,4 +727,291 @@ protected theorem fundamentalInterior : IsFundamentalDomain G (fundamentalInteri
+    rw [h𝓕.projection_respects_measure (μ := μ)]
+
+/-- If a fundamental domain has volume 0, then `QuotientMeasureEqMeasurePreimage` holds. -/
+@[to_additive IsAddFundamentalDomain.addQuotientMeasureEqMeasurePreimage_of_zero]
+theorem IsFundamentalDomain.quotientMeasureEqMeasurePreimage_of_zero
+    {s : Set α} (fund_dom_s : IsFundamentalDomain G s ν)
+    (vol_s : ν s = 0) :
+    QuotientMeasureEqMeasurePreimage ν (0 : Measure (Quotient α_mod_G)) := by
+  apply fund_dom_s.quotientMeasureEqMeasurePreimage ν
+  ext U meas_U
+  simp only [zero_toOuterMeasure, OuterMeasure.coe_zero, Pi.zero_apply]
+  convert (measure_inter_null_of_null_right (h := vol_s) (Quotient.mk α_mod_G ⁻¹' U)).symm
+  rw [measureRestrictMap_apply (meas_U := meas_U)]
+
+/-- If a measure `μ` on a quotient satisfies `QuotientMeasureEqMeasurePreimage` with respect to a
+sigma-finite measure `ν`, then it is itself `SigmaFinite`. -/
+@[to_additive]
+lemma QuotientMeasureEqMeasurePreimage.sigmaFiniteQuotient
+    [i : SigmaFinite ν] [i' : HasFundamentalDomain G α ν]
+    (μ : Measure (Quotient α_mod_G)) [QuotientMeasureEqMeasurePreimage ν μ] :
+    SigmaFinite μ := by
+  rw [sigmaFinite_iff]
+  obtain ⟨A, hA_meas, hA, hA'⟩ := Measure.toFiniteSpanningSetsIn (h := i)
+  simp only [mem_setOf_eq] at hA_meas
+  refine ⟨⟨fun n ↦ π '' (A n), by simp, fun n ↦ ?_, ?_⟩⟩
+  · obtain ⟨s, fund_dom_s⟩ := i'
+    have : π ⁻¹' (π '' (A n)) = _ := MulAction.quotient_preimage_image_eq_union_mul (A n) (G := G)
+    have measπAn : MeasurableSet (π '' A n) := by
+      let _ : Setoid α := α_mod_G
+      rw [measurableSet_quotient, Quotient.mk''_eq_mk, this]
+      apply MeasurableSet.iUnion
+      exact fun g ↦ MeasurableSet.const_smul (hA_meas n) g
+    rw [fund_dom_s.projection_respects_measure_apply (μ := μ) measπAn, this, iUnion_inter]
+    refine lt_of_le_of_lt ?_ (hA n)
+    rw [fund_dom_s.measure_eq_tsum (A n)]
+    exact measure_iUnion_le _
+  · rw [← image_iUnion, hA']
+    refine image_univ_of_surjective (by convert surjective_quotient_mk' α)
+
+/-- A measure `μ` on `α ⧸ G` satisfying `QuotientMeasureEqMeasurePreimage` and having finite
+covolume is a finite measure. -/
+@[to_additive]
+theorem QuotientMeasureEqMeasurePreimage.isFiniteMeasure_quotient
+    (μ : Measure (Quotient α_mod_G)) [QuotientMeasureEqMeasurePreimage ν μ]
+    [hasFun : HasFundamentalDomain G α ν] (h : covolume G α ν ≠ ∞) :
+    IsFiniteMeasure μ := by
+  obtain ⟨𝓕, h𝓕⟩ := hasFun.ExistsIsFundamentalDomain
+  rw [h𝓕.projection_respects_measure (μ := μ)]
+  have : Fact (ν 𝓕 < ∞) := by
+    apply Fact.mk
+    convert Ne.lt_top h
+    exact (h𝓕.covolume_eq_volume ν).symm
+  infer_instance
+
+/-- A finite measure `μ` on `α ⧸ G` satisfying `QuotientMeasureEqMeasurePreimage` has finite
+covolume. -/
+@[to_additive]
+theorem QuotientMeasureEqMeasurePreimage.covolume_ne_top
+    (μ : Measure (Quotient α_mod_G)) [QuotientMeasureEqMeasurePreimage ν μ] [IsFiniteMeasure μ]
+    [hasFun : HasFundamentalDomain G α ν] :
```

Comment:
You shouldn't need this assumption: if there is no fundamental domain, then the covolume is zero...